.github/workflows/compile-requirements.yml

```diff
@@ -0,0 +1,82 @@
+name: Generate requirement files
+
+on:
+  workflow_call:
+    inputs:
+      os:
+        default: '["ubuntu-latest"]'
+        description: A string representation of a list of operating systems.
+        required: false
+        type: string
+      python:
+        default: '["3.10"]'
+        description: A string representation of a list of Python versions.
+        required: false
+        type: string
+      requirements_directory:
+        default: requirements
+        description: The directory where requirements.in is located.
+        required: false
+        type: string
+      requirements_files:
+        default: '["requirements.in"]'
+        description: A string representation of a list of requirement files.
+        required: false
+        type: string
+
+jobs:
+  compile-requirements:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: ${{ fromJSON(inputs.os) }}
+        python-version: ${{ fromJSON(inputs.python) }}
+        requirements_file: ${{ fromJSON(inputs.requirements_files) }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+      - name: Set up Python ${{ matrix.python-version }}
```

Comment:
above it is `fromJSON(inputs.python)`, but here it is `matrix.python-version`?  I would expect these to be the same- My best guess is that the inputs.* is defining the variables for matrix.*.  If that is correct, please confirm.

Reply:
Yes correct -- `inputs` unfortunately can only be passed as strings (from the workflow that is calling this workflow) and we convert them to a list to be used in `matrix` above. Here were are using the value directly from the matrix run.